src/main/java/dev/joopie/jambot/music/command/SearchCommandHandler.java

```diff
@@ -118,28 +118,21 @@ public RestAction<?> handle(final CommandInteraction event) {
 
     @Transactional
     @Override
-    public void onCommandAutoCompleteInteraction(CommandAutoCompleteInteractionEvent event) {
-        if (event.getName().equals(COMMAND_NAME) && event.getFocusedOption().getName().equals(COMMAND_OPTION_INPUT_ARTIST)) {
-            var options = artistRepository.findAll().stream()
+    public List<Command.Choice> autocomplete(CommandAutoCompleteInteraction event) {
+        return switch (event.getFocusedOption().getName()) {
+            case COMMAND_OPTION_INPUT_ARTIST -> artistRepository.findAll().stream()
                     .filter(artist -> artist.getName().startsWith(event.getFocusedOption().getValue())) // only display words that start with the user's current input
                     .map(artist -> new Command.Choice(artist.getName(), artist.getName())) // map the words to choices
+                    .limit(COMMAND_OPTION_MAX_OPTIONS)
                     .toList();
-            event.replyChoices(options).queue();
-        }
-
-        if (event.getName().equals(COMMAND_NAME) && event.getFocusedOption().getName().equals(COMMAND_OPTION_INPUT_SONGNAME)) {
-            var options = trackRepository.findAll().stream()
+            case COMMAND_OPTION_INPUT_SONGNAME -> trackRepository.findAll().stream()
                     .filter(artistTrack -> artistTrack.getArtists().stream()
                             .anyMatch(artist -> artist.getName().contains(event.getOptions().getFirst().getAsString())))
                     .filter(track -> track.getName().startsWith(event.getFocusedOption().getValue()))
                     .map(track -> new Command.Choice(track.getName(), track.getName()))
+                    .limit(COMMAND_OPTION_MAX_OPTIONS)
```

Comment:
Dit is nog steeds problematisch als je `findAll` doet ivm performance.

Nieuw issue maken om `findAll` te vervangen met specifieke query voor autocomplete die niet meer dan 25 resulsts teruggeeft, en rekening houdt met filters.